tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java

```diff
@@ -410,6 +417,34 @@ public Response timeLimit(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
         return Response.status(Response.Status.OK).entity(lraId.toASCIIString()).build();
     }
 
+    /**
+     * @see org.eclipse.microprofile.lra.tck.TckTests#timeLimitWithPreConditionFailed
+     */
+    @GET
+    @Path(TIME_LIMIT_HALF_SEC)
+    @Produces(MediaType.APPLICATION_JSON)
+    @LRA(value = LRA.Type.REQUIRED, timeLimit = 500, timeUnit = ChronoUnit.MILLIS)
+    public Response timeLimitTest2(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
+        assertHeaderPresent(lraId, LRA_HTTP_CONTEXT_HEADER);
+
+        activityStore.add(new Activity(lraId));
+
+        try {
+            Thread.sleep(1000); // sleep for longer than specified in the timeLimit annotation attribute
+            // force the implementation to notice that the LRA should have timed out
+            lraTestService.waitForCallbacks(lraId);
+            // the next request should fail with a 412 code since the LRA should no longer be active
+            restPutInvocation(lraId, MANDATORY_LRA_RESOURCE_PATH, "");
+        } catch (WebApplicationException wae) {
+            return wae.getResponse();
+        } catch (InterruptedException e) {
+            LOGGER.log(Level.FINE, "Interrupted because time limit elapsed", e);
+            fail("Interrupted because time limit elapsed");
```

Comment:
Will this work since this is a JAX-RS resource (i.e. not in the test)

Reply:
Good question, fail() (at least in junit4 master, today) seems to translate into an [exception](https://github.com/junit-team/junit4/blob/master/src/main/java/junit/framework/Assert.java#L57) but I am not sure what the calling side receives if anything?

Reply:
500 I believe since with exception it's failed JAX-RS invocation.

Reply:
I will return a response code instead.